src/gui/remote/DatabaseSettingsWidgetRemote.cpp

```diff
@@ -191,10 +192,24 @@ void DatabaseSettingsWidgetRemote::testDownload()
     }
 
     if (!QFile::exists(result.filePath)) {
-        m_ui->messageWidget->showMessage(tr("Download finished, but file %1 could not be found.").arg(result.filePath),
+        m_ui->messageWidget->showMessage(tr("Command finished, but downloaded file does not exist."),
+                                         MessageWidget::Error);
+        return;
+    }
+
+    if (!hasValidPublicHeaders(result.filePath)) {
+        m_ui->messageWidget->showMessage(tr("Download finished, but file failed KeePass header check. File is not a "
```

Comment:
You could use the actual error from database open here. "KeePass header check" is quite a technical error message as well.

Reply:
I have adapted the error message